File: src/ltv.ts
import { validateCltvInputs, validateLtvInputs } from './validate.js';

/** LTV = loanAmount / propertyValue, as an unrounded decimal ratio (0.9 = 90%). */
export function loanToValue(loanAmount: number, propertyValue: number): number {
  validateLtvInputs(loanAmount, propertyValue);
  return loanAmount / propertyValue;
}

/** CLTV = (firstLienBalance + secondLienBalance) / propertyValue, unrounded decimal ratio. */
export function combinedLoanToValue(
  firstLienBalance: number,
  secondLienBalance: number,
  propertyValue: number,
): number {
  validateCltvInputs(firstLienBalance, secondLienBalance, propertyValue);
  return (firstLienBalance + secondLienBalance) / propertyValue;
}

File: COB-py/cob_calculator/ltv.py
from .validate import validate_cltv_inputs, validate_ltv_inputs


def loan_to_value(loan_amount: float, property_value: float) -> float:
    """LTV = loanAmount / propertyValue, as an unrounded decimal ratio (0.9 = 90%)."""
    validate_ltv_inputs(loan_amount, property_value)
    return loan_amount / property_value


def combined_loan_to_value(
    first_lien_balance: float, second_lien_balance: float, property_value: float
) -> float:
    """CLTV = (firstLienBalance + secondLienBalance) / propertyValue, unrounded ratio."""
    validate_cltv_inputs(first_lien_balance, second_lien_balance, property_value)
    return (first_lien_balance + second_lien_balance) / property_value
